plugins/cake/cake_

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+# -*- sh -*-
+
+: << =cut
+
+=pod
+
+=encoding UTF-8
+
+=head1 NAME
+
+cake_ - Plugin to monitor cake's backlog, dropped, overlimits and requeues
+
+=head1 CONFIGURATION
+
+None needed.
+
+=head1 INTERPRETATION
+
+Cake, also known as sch_cake is a modern bandwidth limiter, which eliminates
+buffer bloat over slow links. It's also capable to give flows, hosts and each 
+flow of each host a fair part of the avaible bandwidth.
+
+This plugin allows for a monitor of the pressure on the qdisc, by monitoring key
+values.
+
+=head1 SEE ALSO
+
+Take a look at "man cake" to get more information about cake.
+
+=head1 MAGIC MARKERS
+
+ #%# family=auto
+ #%# capabilities=autoconf suggest
+
+=head1 AUTHORS
+
+RubenKelevra <ruben@vfn-nrw.de>
+
+work based on the tc plugin, authors:
+Steve Schnepp <steve.schnepp@gmail.com>,
+Samuel Smith <esaym@cpan.org>,
+Nye Liu <nyet@nyet.org>
+
+=head1 LICENSE
+
+GPLv2 or later
+
+=cut
+
+DEVICE=${0##*/cake_}
+
+tc_get_ifb_dev() {
+	dev="$(/sbin/tc filter show dev $DEVICE parent ffff: protocol all | grep "mirred" | grep "Egress Redirect to device")"
```

Comment:
Please use `local` for variables that are only used in functions.
Afterwards you will need to split the assignment from the `local` line in order to not hide the exit code of the command substitution.

Please quote variables (e.g. `$DEVICE`). `shellcheck` is your friend :)

`grep "mirrored"`: `-w` would be more specific here. Or maybe join both `grep` calls into one.

Reply:
Thanks for the heads up, haven't thought about that.

Reply:
To the regex idea: I'm not firm enought to use it here - I don't use stuff I don't fully understand.

Reply:
I *guess* `grep "Egress Redirect to device.*mirrored"` would work here.
(I cannot tell, since my machine does not seem to use cake by default)
Anyway: the current state is also fine.

Reply:
@sumpfralle I doubt that this would work, since the line reads like:

`action order 1: mirred (Egress Redirect to device ifb1) stolen`

mirrored isn't contained ;)

Reply:
I am not sure, if you meant this serious, but I will just follow your invitation for spoon feeding :)
```shell
grep "mirred.*Egress Redirect to device"
```

Additionally this would be a great place for adding a comment with this expected example output line. This will ease the life of future readers of the code.

And I have to admit, that this line of output looks quite human-centric and thus will *really change* somewhen. `json` would be your friend :)

Reply:
;)

And good idea :)